diff --git a/tools/image.c b/tools/image.c
--- a/tools/image.c
+++ b/tools/image.c
@@ -165,6 +165,7 @@ static const table_entry_t uimage_type_names[] = {
 	{	IH_TYPE_FILESYSTEM, "filesystem", "Filesystem Image",	},
 	{	IH_TYPE_FIRMWARE,   "firmware",	  "Firmware",		},
 	{	IH_TYPE_FLATDT,     "flat_dt",    "Flat Device Tree",	},
+	{	IH_TYPE_FLATDT,     "Flat Device Tree", "Flat Device Tree (alias)", },
 	{	IH_TYPE_GPIMAGE,    "gpimage",    "TI Keystone SPL Image",},
 	{	IH_TYPE_KERNEL,	    "kernel",	  "Kernel Image",		},
 	{	IH_TYPE_KERNEL_NOLOAD, "kernel_noload",  "Kernel Image (no loading done)", },
-- 
2.34.1 